### PR TITLE
Deal with non-existent chunks

### PR DESCRIPTION
**Pull request by davidkohn88**
Before this, rows in the source table could have been missed if they
fall outside chunk boundaries (ie if their insertion would cause a new
chunk to be created in the hypertable). This solves this by also
inserting into any gaps in the chunks as well as into the beginning or
end where the chunks may not exist yet.

### CODE REVIEW

**Review by feikesteenbergen — APPROVED** [review]
Haven't tested it, but as far as I can tell from the code, the problem was that we sometimes skipped rows from the source table if no chunk yet existed?

LGTM.

**Comment by davidkohn88** [comment]
Yes. I just modified it to do that in a more efficient way can you take another look and make sure my logic is reasonable?

**Review by feikesteenbergen — APPROVED** [review]
As far as I read it it seems sane, we create those partitions < final chunk in the for loop.

For those rows that > final chunk we have a catchall after the loop.

File: backfill.sql [hunk 1]
```diff
@@ -189,23 +193,35 @@ BEGIN
         -- find the dimension slices that overlap with the data in our staging table 
         -- the range_ends are non inclusive, the range_starts are inclusive
         AND max_time_internal >= ds.range_start AND min_time_internal < ds.range_end
+        ORDER BY ds.range_end
     LOOP
         -- decompress the chunks in the dimension slice, committing transactions after each decompress
         CALL decompress_dimension_slice(dimension_slice_row, chunks_decompressed);
```
Comment (feikesteenbergen) [inline backfill.sql:200]: We could take out an explicit lock on the table here, to prevent the compression job from succeeding?

Comment (davidkohn88) [reply backfill.sql:200]: Hmm...I think you just run into potential deadlock issues.

**Pull request merged**